src/ir/resources.rs

```diff
@@ -1 +1,297 @@
+                property_type,
+                resource_type: &resource.resource_type,
+            };
+
+            let ir = translate_resource(prop, &rt).unwrap();
+            props.insert(name.to_string(), ir);
+        }
+
+        resource_instructions.push(ResourceInstruction {
+            name: resource.name.to_string(),
+            resource_type: resource.resource_type.to_string(),
+            condition: resource.condition.clone(),
+            properties: props,
+        });
+    }
+    resource_instructions
+}
+
+fn translate_resource(
+    resource_value: &ResourceValue,
+    resource_translator: &ResourceTranslationInputs,
+) -> Result<ResourceIr, TransmuteError> {
+    match resource_value {
+        ResourceValue::Null => Ok(ResourceIr::Null),
+        ResourceValue::Bool(b) => Ok(ResourceIr::Bool(*b)),
+        ResourceValue::Number(n) => Ok(ResourceIr::Number(*n)),
+        ResourceValue::String(s) => Ok(ResourceIr::String(s.to_string())),
+        ResourceValue::Array(parse_resource_vec) => {
+            let mut array_ir = Vec::new();
+            for parse_resource in parse_resource_vec {
+                let x = translate_resource(parse_resource, resource_translator)?;
+                array_ir.push(x);
+            }
+
+            Ok(ResourceIr::Array(
+                resource_translator.complexity.clone(),
+                array_ir,
+            ))
+        }
+        ResourceValue::Object(o) => {
+            let mut new_hash = HashMap::new();
+            for (s, rv) in o {
+                let property_ir = match resource_translator.complexity {
+                    Complexity::Simple(_) => translate_resource(rv, resource_translator)?,
+                    Complexity::Complex(_) => {
+                        // Update the rule with it's underlying property rule.
+                        let mut new_rt = resource_translator.clone();
+                        let rule = resource_translator
+                            .specification
+                            .property_types
+                            .get(&resource_translator.property_type.unwrap().to_string())
+                            .unwrap();
+                        let properties = rule.properties.as_ref().unwrap();
+                        let property_rule = properties.get(s).unwrap();
+                        new_rt.complexity = property_rule.get_complexity();
+
+                        let opt = match property_rule.get_complexity() {
+                            Complexity::Simple(_) => Option::None,
+                            Complexity::Complex(x) => {
+                                let mut ret =
+                                    format!("{}.{}", resource_translator.resource_type, x);
+                                if x == "Tag" {
+                                    ret = "Tag".to_string();
+                                }
+                                Option::Some(ret)
+                            }
+                        };
```

Comment:
Worth moving to its own method since it's re-used in two places?

Reply:
Yeah. I'ma push so we got a working application but will then update to clean this up.